### PR TITLE
Package up python dependencies into Lambda zip

### PR DESCRIPTION
**Pull request by abrooksv**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Description
Bundles up 3rd party dependencies from site-packages while keeping the virtual env out of the lambda zip

## Motivation and Context
Allows packaging of 3rd party dependencies into Lambda zip

## Related Issue(s)
#285

## Testing
Manually tested based on sam init template, new tests

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] Local run of `gradlew check` succeeds
- [x] My code follows the code style of this project
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/python/PythonLambdaPackager.kt [hunk 1]
```diff
@@ -21,36 +24,47 @@ import java.util.concurrent.CompletionStage
...
-                                VfsUtilCore.getRelativeLocation(
-                                    file,
-                                    contentRoot
-                                )!!, fileContents
-                            )
+                    ModuleRootManager.getInstance(module).contentRoots.forEach { contentRoot ->
+                        contentRoot.walkFiles(excludedRoots) { file ->
+                            mappings[contentRoot.path] = "/"
+                            file.inputStream.use { fileContents ->
+                                zip.putNextEntry(VfsUtilCore.getRelativeLocation(file, contentRoot)!!, fileContents)
```
Comment (kiiadi) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/python/PythonLambdaPackager.kt]: I hate bang-bangs (`!!`) are we confident we're handling the null properly?

File: jetbrains-core/src/software/aws/toolkits/jetbrains/utils/ZIpUtils.kt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.utils
```
Comment (kiiadi) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/utils/ZIpUtils.kt]: 1) `ZIpUtils.kt` - the i should be lower-case
2) Don't we already have zip utils: `software.aws.toolkits.core.utils.ZipUtils`

Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/utils/ZIpUtils.kt]: Fixed

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/python/PythonLambdaPackager.kt (hunk 1 above)
Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/python/PythonLambdaPackager.kt]: Can't hurt to wrap it

**Pull request merged**